engine/src/main/java/com/datatorrent/stram/StramLocalCluster.java

```diff
@@ -283,21 +281,26 @@ public StramLocalCluster(LogicalPlan dag) throws IOException, ClassNotFoundExcep
     dag.validate();
     // ensure plan can be serialized
     cloneLogicalPlan(dag);
-    // convert to URI so we always write to local file system,
-    // even when the environment has a default HDFS location.
-    String pathUri = CLUSTER_WORK_DIR.toURI().toString();
+    final String pathUri;
+    String appPath = dag.getAttributes().get(LogicalPlan.APPLICATION_PATH);
+    if (appPath == null) {
+      // convert to URI so we always write to local file system,
+      // even when the environment has a default HDFS location.
+      // assumes execution as unit test
+      pathUri = new File("target", StramLocalCluster.class.getName()).toURI().toString();
```

Comment:
I think we shouldn't make the "target/<classname>" as the default app path in the main code. This assumes that we are running a unit test, but I think the unit tests that run the code should set this explicitly. I think the default for local mode should be in /tmp